vgscollect/src/main/java/com/verygoodsecurity/vgscollect/view/InputFieldView.kt

```diff
@@ -774,11 +778,32 @@ abstract class InputFieldView @JvmOverloads constructor(
             inputField.setTextAppearance(context, textAppearance)
         }
 
-        val bgDraw = background?.constantState?.newDrawable()
-        if (bgDraw != null) {
-            inputField.background = bgDraw
+        background?.constantState?.newDrawable()?.apply {
+            setBackgroundColor(Color.TRANSPARENT)
+            bgDraw = this
+            inputField.background = this
         }
-        setBackgroundColor(Color.TRANSPARENT)
+    }
+
+    private var bgDraw: Drawable? = null
+
+    override fun setBackgroundColor(color: Int) {
+        background = ColorDrawable(color)
+    }
+
+    override fun setBackground(background: Drawable?) {
+        when {
+            ::inputField.isInitialized || bgDraw != null -> {
```

Comment:
@Endorf I think here can be the exception, if for example `inputField.isInitialized` == false and `bgDraw != null == true`, in this case` inputField.background = background` will throws exception.